### PR TITLE
lookup JSONSerializer instance through store instead of manual instan…

### PR DESCRIPTION
**Pull request by fivetanley**
…tiation

### CODE REVIEW

File: packages/ember-data/lib/system/model/model.js [hunk 1]
```diff
@@ -375,7 +374,7 @@ var Model = Ember.Object.extend(Ember.Evented, {
   */
   toJSON: function(options) {
     // container is for lazy transform lookups
-    var serializer = JSONSerializer.create({ container: this.container });
+    var serializer = this.store.serializerFor('-default');
```
Comment (sandstrom) [inline packages/ember-data/lib/system/model/model.js]: Would it be possible to do something like `store.serializerFor(this.constructor.typeKey)` instead? To get the record-specific serializer when one is available.

We've had this work-around for a long time (code not aligned with recent ED, but it shows the main idea):
```
toJSON: function(options) {
  var serializer = this.store.serializerFor(this.constructor.typeKey);
  return serializer.serialize(this, options);
}
```

Comment (igorT) [inline packages/ember-data/lib/system/model/model.js]: no, if you want the record specific serializer, you `record.serialize()`. `toJSON` is using the json serializer for debugging and similar purposes

Comment (sly7-7) [inline packages/ember-data/lib/system/model/model.js]: I have a doubt, is the '-default' serializer always JsonSerializer ? Is'nt it possible to override this ?

Comment (sandstrom) [inline packages/ember-data/lib/system/model/model.js]: You are right, thanks! This part of the code is old, I see now that the issue I worked around has been solved for ~2 years :smile:

**Pull request merged**